MetaMask/metamask-extension
# feat: UX: Multichain: Show in-page toast when chain changed

## **Description**

This PR implements an in-page toast when the network changes.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23834?quickstart=1)

## **TO DO**
- [ ] - Add animation / transition
- [x] - Update duration
- [x] - Cut down on noisiness of `ethereum_switchChain`
- [x] - Add E2E test

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2334

## **Manual testing steps**

1. Go to any dapp
2. Change networks
3. See the in-page toast

## **Screenshots/Recordings**



### **Before**

N/A

### **After**


https://github.com/MetaMask/metamask-extension/assets/46655/084e508a-0cee-4068-be4e-aa95dbe2ced0




## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
